### PR TITLE
Kiel RecyclerView.Adapter Builder integration

### PR DESCRIPTION
**Pull request by ibrahimyilmaz**
Hi @rizmaulana 

I would like to contribute your project by integrating Kiel(https://github.com/ibrahimyilmaz/kiel).

Any feedback is greatly welcome

Looking forward to hearing from you
Sincerely
Ibra

### CODE REVIEW

File: app/src/main/java/id/rizmaulana/covid19/ui/dailygraph/DailyDataListFragment.kt [hunk 1]
```diff
@@ -18,9 +16,8 @@ class DailyDataListFragment : BaseFragment() {
     private val viewModel by sharedViewModel<DailyGraphViewModel>()
 
     lateinit var binding: FragmentDailyListBinding
-    private val dailyAdapter by lazy {
-        VisitableRecyclerAdapter(DailyFactoryImpl())
-    }
+
+    private val dailyAdapter = createAdapter()
```
Comment (rizmaulana) [inline app/src/main/java/id/rizmaulana/covid19/ui/dailygraph/DailyDataListFragment.kt]: Please use lazy init

File: app/src/main/java/id/rizmaulana/covid19/ui/detail/DetailActivity.kt [hunk 2]
```diff
@@ -49,9 +48,9 @@ class DetailActivity : BaseActivity() {
         }
     }
 
-    private val detailAdapter by lazy {
-        VisitableRecyclerAdapter(ItemTypeFactoryImpl(), ::onItemClick, ::onLongItemClick)
-    }
+    private val detailAdapter = createAdapter(
```
Comment (rizmaulana) [inline app/src/main/java/id/rizmaulana/covid19/ui/detail/DetailActivity.kt]: Please use lazy init

File: app/src/main/java/id/rizmaulana/covid19/ui/overview/DashboardActivity.kt [hunk 3]
```diff
@@ -24,12 +23,9 @@ import org.koin.android.viewmodel.ext.android.viewModel
 class DashboardActivity : BaseActivity() {
 
     private val viewModel by viewModel<DashboardViewModel>()
-    private val dailyAdapter by lazy {
-        VisitableRecyclerAdapter(
-            ItemTypeFactoryImpl(),
-            ::onItemClicked
-        )
-    }
+    private val dailyAdapter = createAdapter(
```
Comment (rizmaulana) [inline app/src/main/java/id/rizmaulana/covid19/ui/overview/DashboardActivity.kt]: Please use lazy init

File: app/src/main/java/id/rizmaulana/covid19/ui/percountry/indonesia/CountryIndonesiaActivity.kt [hunk 4]
```diff
@@ -18,12 +17,7 @@ import org.koin.android.viewmodel.ext.android.viewModel
 class CountryIndonesiaActivity : BaseActivity() {
     private val viewModel by viewModel<CountryIndonesiaViewModel>()
     private lateinit var binding: ActivityCountryIndonesiaBinding
-    private val viewAdapter by lazy {
-        VisitableRecyclerAdapter(
-            ItemTypeFactoryImpl(),
-            ::onItemClicked
-        )
-    }
+    private val viewAdapter = createAdapter(::onItemClicked)
```
Comment (rizmaulana) [inline app/src/main/java/id/rizmaulana/covid19/ui/percountry/indonesia/CountryIndonesiaActivity.kt]: Please use lazy init

**Review by rizmaulana — COMMENTED** [review]
Just little comment.

Btw cool lib 👍

**Comment by ibrahimyilmaz** [comment]
I will make them lazy :) thank you for your review.

**Comment by ibrahimyilmaz** [comment]
They all are now lazy!

**Review by rizmaulana — APPROVED** [review]
LGTM!

**Pull request merged**